AGENTS.md

```diff
@@ -0,0 +1,952 @@
+
+### 3. **Host Helpers**
+
+Mixins that provide common functionality for Satellite and ContentHost objects.
+
+**Satellite Mixins** (`robottelo/host_helpers/satellite_mixins.py`):
+- `api_factory`: API entity creation methods
+- `cli_factory`: CLI entity creation methods
+- `ui_session()`: Context manager for UI sessions
+
+**ContentHost Mixins** (`robottelo/host_helpers/contenthost_mixins.py`):
+- `register_contenthost()`: Register host to Satellite
+- `install_katello_ca()`: Install Katello CA certificate
+- `execute()`: Run commands on content host
+
+Example:
+
+```python
+# Using Satellite API factory
+ak = target_sat.api.ActivationKey(organization=org).create()
+
+# Using UI session
+with target_sat.ui_session() as session:
+    session.activationkey.create({'name': 'my-ak'})
+
+# Using ContentHost methods
+rhel_contenthost.register_contenthost(org, ak)
+result = rhel_contenthost.execute('subscription-manager status')
+```
+
+### 4. **Data Factories**
+
+Generate random test data using `robottelo.utils.datafactory`.
+
+**Common Functions**:
+- `gen_string()`: Generate random strings
+- `gen_alpha()`: Generate alphabetic strings
+- `gen_numeric_string()`: Generate numeric strings
+- `gen_email()`: Generate email addresses
+
+Example:
+
+```python
+from robottelo.utils.datafactory import gen_string
+
+name = gen_string('alpha', 10)  # Random 10-char alphabetic string
+email = gen_email()  # Random email
+```
+---
+
+## Code Standards
+
+### Import Ordering
+
+1. **Standard library** imports
+2. **Third-party** imports (alphabetical)
+3. **Robottelo** imports (alphabetical)
+4. Blank line between groups
+
+```python
+# Standard library
+import logging
+from datetime import datetime
+
+# Third-party
+import pytest
+from box import Box
+from nailgun.entities import ActivationKey
+
+# Robottelo
+from robottelo.config import settings
+from robottelo.constants import DEFAULT_CV
+from robottelo.utils.datafactory import gen_string
+```
+
+### Naming Conventions
+
+| Type | Convention | Example |
+|------|-----------|---------|
+| **Test Functions** | `test_{type}_{action}_{entity}` | `test_positive_create_activation_key` |
+| **Fixtures** | `{scope}_{entity}` | `module_activation_key`, `function_org` |
+| **Classes** | PascalCase | `ActivationKey`, `ContentView` |
+| **Functions/Methods** | snake_case | `create()`, `delete()`, `register_contenthost()` |
+| **Constants** | UPPER_SNAKE_CASE | `DEFAULT_TIMEOUT`, `OPENSSH_RECOMMENDATION` |
+| **Private** | Leading underscore | `_helper()`, `_validate()` |
+
+### Test Naming Pattern
+
+Test names follow a specific pattern to indicate expected behavior:
+
+- `test_positive_*`: Test should succeed (happy path)
+- `test_negative_*`: Test should fail with expected error (error handling)
+- `test_upgrade_*`: Upgrade scenario test
+
+Examples:
+- `test_positive_create_activation_key_with_cv()`
+- `test_negative_create_ak_with_invalid_name()`
+- `test_upgrade_content_view_promotion()`
+
+### Docstring Style
```

Comment:
Here I'd point the agent to `testimony.yaml` so it knows what the choices for each [token](https://github.com/SatelliteQE/testimony/blob/master/docs/index.rst#test-case-docstring-format) are. `testimony.yaml` can also be the source of truth for what tokens are required.